### ISSUE TITLE
SA1313 should not be reported when method implements a interface

### ISSUE THREAD

**Issue by drieseng** [DotNetAnalyzers/StyleCopAnalyzers#3555]
SA1313 should not be reported when a given method implements an interface method with parameter names that do not start with a lowercase letter.

For example:
```c#
public interface IDTExtensibility2
{
    // SA1313 is reported here, this is fine
    void OnStartupComplete(ref Array Custom);

    // SA1313 is reported here, this is fine
    void OnBeginShutdown(ref Array Custom);
}

public class AddIn : IDTExtensibility2
{
    // SA1313 should not be reported here
    void IDTExtensibility2.OnStartupComplete(ref Array Custom)
    {
    }

    // SA1313 should not be reported here
    public void OnBeginShutdown(ref Array Custom)
    {
    }
}
```

**Comment by bjornhellander** [comment]
Grabbing this

**Comment by bjornhellander** [comment]
@drieseng, can you confirm that the only SA1313 you get **inside AddIn** is for OnStartupComplete? That is at least the behaviour that I see. So you have a single unexpected SA1313 in your example?

**Comment by drieseng** [comment]
@bjornhellander Indeed, **SA1313** is only falsely reported for an explicit interface implementation. For the public method (`OnBeginShutdown`, in my example) it's already ok.

Thanks for the PR!
